robot/impl/local_robot_test.go

```diff
@@ -3420,3 +3280,16 @@ func TestCloudMetadata(t *testing.T) {
 		})
 	})
 }
+
+func initTestRobot(t *testing.T, cfg *config.Config) (robot.LocalRobot, *observer.ObservedLogs, func()) {
+	t.Helper()
+
+	ctx := context.Background()
+	logger, logs := rtestutils.NewInfoObservedTestLogger(t)
```

Comment:
Alternatively, we can make it so that you still need to pass in the logger and/or context. The main thing I want to get rid of the boilerplate error checking and deferred close.

Reply:
See https://github.com/viamrobotics/rdk/pull/3605/files#r1500956774 for why I made this `INFO` level